src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs

```diff
@@ -741,10 +937,14 @@ public void ChangeConnectionDatabaseContext(string ownerUri, string newDatabaseN
             {
                 try
                 {
-                    if (info.SqlConnection.State == ConnectionState.Open)
+                    foreach (DbConnection connection in info.GetAllConnections())
                     {
-                        info.SqlConnection.ChangeDatabase(newDatabaseName);
+                        if (connection.State == ConnectionState.Open)
```

Comment:
Ok, if it's not new, then I'd say it's fine to leave it be. But we should remember this for the future just in case.